packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -449,210 +528,170 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
+                  accounts[2].wallet.address,
+                  requestDetails,
+                );
+                hbarSpendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(
+                  ethSpendingPlan.planId,
+                  requestDetails,
+                );
+              }
+              const amountSpent = await pollForProperAmountSpent(hbarSpendingPlan, deploymentCounts, expectedTxCost);
               const remainingHbarsAfter = Number(await metrics.get(testConstants.METRICS.REMAINING_HBAR_LIMIT));
 
               // Explanation:
               // An HBAR limit check triggers the HBAR_RATE_LIMIT_EXCEED error in two scenarios:
-              //    a. if remainingHbarsBefore > maxBasicSpendingLimit ===> (totalHbarSpentByBasicPlan + estimatedTxFee) > maxBasicSpendingLimit
-              //    b. if remainingHbarsBefore <= maxBasicSpendingLimit ===> (remainingBudget - estimatedTxFee) < 0
+              //    a. if remainingHbarsBefore > maxBasicSpendingLimit ===> (totalHbarSpentByBasicPlan + expectedTxCost) > maxBasicSpendingLimit
+              //    b. if remainingHbarsBefore <= maxBasicSpendingLimit ===> (remainingBudget - expectedTxCost) < 0
               if (remainingHbarsBefore > maxBasicSpendingLimit) {
-                expect(amountSpent + estimatedTxFee).to.be.gt(maxBasicSpendingLimit);
+                expect(amountSpent + expectedTxCost).to.be.gt(maxBasicSpendingLimit);
               } else {
-                expect(remainingHbarsAfter).to.be.lt(estimatedTxFee);
+                expect(remainingHbarsAfter).to.be.lt(expectedTxCost);
               }
             }
           });
         });
 
         describe('@hbarlimiter-batch2 Preconfigured Tiers', () => {
-          const createAliasForNonBasicPlans = async (subscriptionTier: SubscriptionTier) => {
-            const aliasAccount = await Utils.createAliasAccount(
-              mirrorNode,
-              global.accounts[0],
-              requestId,
-              initialBalance,
-            );
-
-            const hbarSpendingPlan = await hbarSpendingPlanRepository.create(subscriptionTier, requestDetails, mockTTL);
-
-            await ethAddressSpendingPlanRepository.save(
-              { ethAddress: aliasAccount.address, planId: hbarSpendingPlan.id },
-              requestDetails,
-              mockTTL,
-            );
-
-            const plan = await ethAddressSpendingPlanRepository.findByAddress(aliasAccount.address, requestDetails);
-            expect(plan.ethAddress).to.eq(aliasAccount.address);
-            expect(plan.planId).to.eq(hbarSpendingPlan.id);
-            const spendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(plan.planId, requestDetails);
-            expect(spendingPlan.active).to.be.true;
-            expect(spendingPlan.amountSpent).to.eq(0);
-            expect(spendingPlan.subscriptionTier).to.eq(subscriptionTier);
-
-            return { aliasAccount, hbarSpendingPlan };
-          };
```

Comment:
**Note for reviewer:** 
This function was moved to a more general location and renamed `createAliasAndAssociateSpendingPlan()` to enable reuse in other tests.